controllers/kubeadmconfig_controller.go

```diff
@@ -50,14 +50,20 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 
 	config := kubeadmv1alpha2.KubeadmConfig{}
 	if err := r.Get(ctx, req.NamespacedName, &config); err != nil {
-		log.Error(err, "failed to get config", "stacktrace", fmt.Sprintf("%+v", err))
+		log.Error(err, "failed to get config")
 		return ctrl.Result{}, err
 	}
 
+	// bail super early if it's already ready
+	if config.Status.Ready {
+		log.Info("ignoring an already ready config")
+		return ctrl.Result{}, nil
+	}
+
 	// Find the owner reference
 	var machineRef *v1.OwnerReference
 	for _, ref := range config.OwnerReferences {
-		if ref.Kind == machineKind {
+		if fmt.Sprintf("%s, Kind=%s", ref.APIVersion, ref.Kind) == machineKind {
```

Comment:
hmm, refs don't have a GroupVersionKind though, what am I comparing machineKind with? Wouldn't I have to build it up from the Ref parts anyway?

Reply:
My bad (I'll blame it on being sick 😄). I think you can use CAPI's `util.HasOwner(config.OwnerReferences, v1alpha2.SchemeGroupVersion.String(), "Machine")` instead of this loop. Will that do what you need?

Reply:
that seems like a heavy function for this when I just want to compare two strings, does
`ref.Kind == machineKind.Kind && ref.APIVersion == machineKind.Version` look ok to you? It passes the tests.

The only reason i'm a little hesitant on that function is that it doesn't make the code easier to read whereas i know what's going on quickly comparing a ref and a version.